operators/pkg/tenant-controller/keycloak_utils.go

```diff
@@ -65,3 +71,147 @@ func deleteKcRoles(ctx context.Context, kcA *KcActor, rolesToDelete []string) er
 	}
 	return nil
 }
+
+func (kcA *KcActor) getUserInfo(ctx context.Context, username string) (*string, *string, error) {
+
+	usersFound, err := kcA.Client.GetUsers(ctx, kcA.Token.AccessToken, kcA.TargetRealm, gocloak.GetUsersParams{Username: &username})
+	if err != nil {
+		klog.Errorf("Error when trying to find user %s", username)
+		klog.Error(err)
+		return nil, nil, err
+	} else if len(usersFound) == 0 {
+		// no existing users found, create a new one
+		return nil, nil, nil
+	} else if len(usersFound) == 1 {
+		return usersFound[0].ID, usersFound[0].Email, nil
+	} else if len(usersFound) > 1 {
+		klog.Error("Found too many users")
+		return nil, nil, errors.New("FOund too many users")
+	}
+	return nil, nil, fmt.Errorf("Error when getting user %s", username)
+}
+
+func (kcA *KcActor) createKcUser(ctx context.Context, username string, firstName string, lastName string, email string) (*string, error) {
+	tr := true
+	fa := false
+	newUser := gocloak.User{
+		Username:      &username,
+		FirstName:     &firstName,
+		LastName:      &lastName,
+		Email:         &email,
+		Enabled:       &tr,
+		EmailVerified: &fa,
+	}
+	newUserID, err := kcA.Client.CreateUser(ctx, kcA.Token.AccessToken, kcA.TargetRealm, newUser)
+	if err != nil {
+		klog.Errorf("Error when creating user %s", username)
+		klog.Error(err)
+		return nil, err
+	}
+	// 30 days (in seconds) life span for email action
+	if err = kcA.Client.ExecuteActionsEmail(ctx, kcA.Token.AccessToken, kcA.TargetRealm, gocloak.ExecuteActionsEmail{
+		UserID:   &newUserID,
+		Lifespan: &kcA.EmailActionsLifeSpanS,
+		Actions:  &kcA.UserRequiredActions,
+	}); err != nil {
+		klog.Errorf("Error when sending email actions for user %s", username)
+		klog.Error(err)
+		return nil, err
+	}
+
+	klog.Infof("Sent verification email to user %s", username)
```

Comment:
Do you plan to add some feedback in the Tenant resource status?

Reply:
At the moment no, we'll need to think how to handle the resending of a verification email if the user forgots to confim before the timeout